src/basic/socket-util.c

```diff
@@ -239,27 +239,32 @@ int socket_address_parse_and_warn(SocketAddress *a, const char *s) {
 int socket_address_parse_netlink(SocketAddress *a, const char *s) {
         int family;
         unsigned group = 0;
-        _cleanup_free_ char *sfamily = NULL;
+        const char *p, *sfamily = s;
+        int r;
+
         assert(a);
         assert(s);
 
-        zero(*a);
-        a->type = SOCK_RAW;
+        p = strchr(s, ' ');
+        if (p) {
+                sfamily = strndupa(s, p - s);
 
-        errno = 0;
-        if (sscanf(s, "%ms %u", &sfamily, &group) < 1)
```

Comment:
I think it is better than `ifdef` and neater than the way I suggested earlier.

`sscanf` skips leading whitespaces, so in theory the new implementation might break something if the callers depend on that behavior (but I hope they don't :-)).

Reply:
Eh, I tried to add the skipping of spaces, but then the code becomes quite ugly. The version with `sscanf` is nicer after all.

So instead I tried a different approach: filter out ListenNetlink= in the fuzzer. This keeps the work-around localized to the fuzzing code.